roofit/roofitcore/src/RooAbsReal.cxx

```diff
@@ -2605,23 +2605,36 @@ double RooAbsReal::getPropagatedError(const RooFitResult &fr, const RooArgSet &n
   RooArgSet allParamsInAbsReal;
   getParameters(&nset, allParamsInAbsReal);
 
-  // Strip out parameters with zero error
   RooArgList paramList;
-  for(auto * rrvInFitResult : static_range_cast<RooRealVar*>(fr.floatParsFinal())) {
-     if (rrvInFitResult->getError() > 1e-20) {
-        auto * rrvInAbsReal = static_cast<RooRealVar*>(allParamsInAbsReal.find(*rrvInFitResult));
-
-        if(rrvInAbsReal->getVal() != rrvInFitResult->getVal()) {
-           throw std::runtime_error(
-                   std::string("RooAbsReal::getPropagatedError(): the parameters of the RooAbsReal don't have") +
-                   "the same values as in the fit result! The logic of getPropagatedError is broken in this case.");
-        }
+  for(auto * rrvFitRes : static_range_cast<RooRealVar*>(fr.floatParsFinal())) {
+
+     auto rrvInAbsReal = static_cast<RooRealVar const*>(allParamsInAbsReal.find(*rrvFitRes));
+
+     // If this RooAbsReal is a RooRealVar in the fit result, we don't need to
+     // propagate anything and can just return the error in the fit result
+     if(rrvFitRes->namePtr() == namePtr()) return rrvFitRes->getError();
+
+     // Strip out parameters with zero error
+     if (rrvFitRes->getError() <= 1e-20) continue;
```

Comment:
why hard coding here 1.E-20 ? Maybe use numerical precision epsilon ?

Reply:
Yes, that is indeed a better choice than this magic number. I have updated it here and also in another place in that file